src/olympia/amo/urlresolvers.py

```diff
@@ -77,11 +78,23 @@ def get_app(self):
 
     def get_language(self):
         """
-        Return a locale code that we support on the site using the
-        user's Accept Language header to determine which is best.  This
-        mostly follows the RFCs but read bug 439568 for details.
+        Return a locale code that we support on the site using `lang` from GET
+        or POST, falling back to the user's Accept Language header if
+        necessary (mostly following the RFCs but read bug 439568 for details).
         """
-        data = self.request.GET or self.request.POST
+        try:
+            data = self.request.GET or self.request.POST
```

Comment:
Not strictly related to this issue/patch but I didn't think we supported `lang` setting the language as POST data- only as a url parameter?  If it's the `request.POST` that raises then only checking `request.GET` would mean we wouldn't need this handling.

Reply:
Actually even outside of storing translations, having a way for a POST API request to specify the language outside of just `Accept-Language` header might be useful... it would be used to translate error messages for instance

Reply:
... but we do that through the query string, i.e. for ratings we `POST` to `https://addons.mozilla.org/api/v5/ratings/rating/?lang=<lang>`, and the language is not part of the post data. So we could probably remove this...

Reply:
There are writeable apis available under v3/v4 that have translations but they were all internal auth (collections, accounts) - it's unclear if anyone used them with the session cookie and flat strings. 

We don't appear to document how you're supposed to set the "current" language in v3/v4 where we mention you can specify a single flat string for localized values so we don't know if people would have thought to specify it in the POST or accept language header (or as a URL param)

Reply:
There are 2 kind of API error messages: those tied to the client misusing parameters in some way (invalid or missing parameters) and those tied to user-supplied data. The former we tend not to translate, but I would expect us to always translate the latter.

That being said, we support and use `lang` from the query string even in POSTS as mentioned above, we support `Accept-Language`, and we haven't really documented (or used ourselves) that behavior where we look for `lang` in `POST` data, so I think it's fine to remove. I'll do that.